agent/fw_laravel_queue.c

```diff
@@ -624,8 +579,52 @@ NR_PHP_WRAPPER(nr_laravel_queue_workcommand_fire) {
    * aren't executed if we're not actually in a transaction.
    *
    * So instead, what we'll do is to keep recording, but ensure that we ignore
-   * the transaction after WorkCommand::fire() has finished executing, at which
-   * point no more jobs can be run.
+   * the transaction after WorkCommand::handle() has finished executing, at
+   * which point no more jobs can be run.
+   */
+
+  /*
+   * Start listening for jobs.
+   */
+  nr_php_wrap_user_function_before_after_clean(
```

Comment:
Is there a reason why we are setting this wrapper here, instead of in `nr_laravel_queue_enable`? This creates a non-transient wrapper, so by next request, any context gained by not setting it in `nr_laravel_queue_enable` is almost certainly lost.

If the goal is to not instrument `Illuminate\Queue\Worker::process` unless within a call to `Illuminate\Queue\Console\WorkCommand::handle`, perhaps a global boolean that holds that context would be better? (similar to the check_cufa added here https://github.com/newrelic/newrelic-php-agent/pull/729/files)

Reply:
No need to rewrap the non-transient wrapper.
We can wrap both at once, since they are both used together; it doesn't make sense to have one without the other.`WorkCommand::handle` is encountered, `Worker::Process` will be in the calling chain for async queues.

Please see: [25c1652](https://github.com/newrelic/newrelic-php-agent/pull/746/commits/25c16523602c113c894d568f45902bd028337d94)

Reply:
Both `WorkCommand::handle` and `Worker::Process` are no longer used for PHP8+.